packages/web-components/src/tree-item/tree-item.ts

```diff
@@ -0,0 +1,167 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import { attr, customElement, FASTElement, observable, volatile } from '@microsoft/fast-element';
+import { FluentDesignSystem } from '../fluent-design-system.js';
+import { template } from './tree-item.template.js';
+import { styles } from './tree-item.style.js';
+
+@customElement({
+  name: `${FluentDesignSystem.prefix}-tree-item`,
+  template,
+  styles,
+})
+export class TreeItem extends FASTElement {
+  /**
+   * When true, the control will be immutable by user interaction. See {@link https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled | disabled HTML attribute} for more information.
+   * @public
+   * @remarks
+   * HTML Attribute: disabled
+   */
+  @attr({ mode: 'boolean' })
+  disabled = false;
+
+  /**
+   * When true, the control will be appear expanded by user interaction.
+   * @public
+   * HTML Attribute: expanded
+   */
+  @attr({ mode: 'boolean' })
+  expanded = false;
+  protected expandedChanged() {
+    if (this.$fastController.isConnected) {
+      this.$emit('expanded-change', this);
+    }
+  }
+
+  /**
+   * When true, the control will appear selected by user interaction.
+   * @public
+   * @remarks
+   * HTML Attribute: selected
+   */
+  @attr({ mode: 'boolean' })
+  selected = false;
+  protected selectedChanged(): void {
+    if (this.$fastController.isConnected) {
+      this.$emit('selected-change', this);
+    }
+  }
+
+  /**
+   * The size of the tree item element
+   */
+  @observable
+  size: 'medium' | 'small' = 'medium';
```

Comment:
Can we turn these string union types into a const object which can be exported from the package for use in `.ts`?